### PR TITLE
Feature/multiple queued fc flat

### PR DESCRIPTION
**Pull request by mejango**
this is a version of #1 that removes unused contract versions, with clear diffs based on JBController3_1

### CODE REVIEW

**Review by xBA5ED — APPROVED** [review]
Comfy cleanup :ship:

File: forge_tests/helpers/TestBaseWorkflow.sol [hunk 1]
```diff
@@ -1,385 +1,318 @@
...
+
+  function jbToken() internal view returns (JBToken) {
+    return _jbToken;
+  }
+
+  function jbLibraries() internal view returns (AccessJBLib) {
+    return _accessJBLib;
+  }
+
+  function getDefaultCycleConfig(
```
Comment (simplemachine92) [inline forge_tests/helpers/TestBaseWorkflow.sol:215]: Can't use this after access constraints pr is merged, maybe it's removed there but idr.

**Comment by simplemachine92** [comment]
FundingCycleStore / tests are cleaner and coverage is there. lgtm 🛸

File: contracts/JBReconfigurationBufferBallot.sol [hunk 2]
```diff
@@ -37,7 +37,13 @@ contract JBReconfigurationBufferBallot is ERC165, IJBFundingCycleBallot {
...
     unchecked {
       // If there was sufficient time between configuration and the start of the cycle, it is approved. Otherwise, it is failed.
-      return (_start - _configured < duration) ? JBBallotState.Failed : JBBallotState.Approved;
+      // If the ballot hasn't yet started, it's state is ApprovalExpected.
+      return
+        (_start - _configured < duration)
+          ? JBBallotState.Failed
+          : (block.timestamp < _start - duration)
+          ? JBBallotState.ApprovalExpected
+          : JBBallotState.Approved;
```
Comment (drgorillamd) [inline contracts/JBReconfigurationBufferBallot.sol:46]:
```suggestion
          : (block.timestamp < _start - duration)
              ? JBBallotState.ApprovalExpected
              : JBBallotState.Approved;
```

File: contracts/JBReconfigurationBufferBallot.sol [hunk 3]
```diff
@@ -37,7 +37,13 @@ contract JBReconfigurationBufferBallot is ERC165, IJBFundingCycleBallot {
 
     unchecked {
       // If there was sufficient time between configuration and the start of the cycle, it is approved. Otherwise, it is failed.
-      return (_start - _configured < duration) ? JBBallotState.Failed : JBBallotState.Approved;
+      // If the ballot hasn't yet started, it's state is ApprovalExpected.
+      return
+        (_start - _configured < duration)
+          ? JBBallotState.Failed
+          : (block.timestamp < _start - duration)
```
Comment (drgorillamd) [inline contracts/JBReconfigurationBufferBallot.sol:44]: cf https://discord.com/channels/775859454780244028/786985806396784690/1173612390710001694
should include a constructor check duration < block.timestamp (or just uncheck this block)

File: contracts/JBReconfigurationBufferBallot.sol [hunk 4]
```diff
@@ -37,7 +37,13 @@ contract JBReconfigurationBufferBallot is ERC165, IJBFundingCycleBallot {
...
+          ? JBBallotState.ApprovalExpected
```
Comment (drgorillamd) [inline contracts/JBReconfigurationBufferBallot.sol:45]: nit: indent

File: forge_tests/TestReconfigure.sol [hunk 5]
```diff
@@ -565,4 +623,181 @@ contract TestReconfigureProject_Local is TestBaseWorkflow {
         assertEq(fundingCycle.number, 2);
         assertEq(fundingCycle.weight, _dataReconfiguration.weight);
     }
+
+    function testMixedStarts() public {
```
Comment (drgorillamd) [inline forge_tests/TestReconfigure.sol:627]: Maybe we should have a test for a single of reconfiguring while fuzzing a ballot delay + start at, compared to configure (ie correct ballot status?), as we introduce changes in ballot state logic

**Pull request merged**